integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java

```diff
@@ -444,6 +444,19 @@ public void testWeek() throws IOException {
     week("2000-01-01", 2, 52);
   }
 
+  @Test
+  public void testWeek_Of_Year() throws IOException {
```

Comment:
Please, add a test which validates that `week` and `week_of_year` returns same result. For better coverage use an index (e.g. calcs) and try different modes/columns.

Reply:
Added a placeholder for now. Below this on line 462 I've added the test `testWeekAlternateSyntaxesReturnTheSameResults()`. I would like to still add more cases to test and compare the `week` and `week_of_year` functions, but before I do that I had a question. Is there a better way to compare the output? On [PR 165](https://github.com/Bit-Quill/opensearch-project-sql/pull/165#discussion_r1026990599) before you suggested comparing with 
```
assertTrue(result1.similar(result2));
```
, but that does not work in this case. It instead returns an error that doesn't really describe what went wrong.
```
REPRODUCE WITH: ./gradlew ':integ-test:integTest' --tests "org.opensearch.sql.sql.DateTimeFunctionIT.testWeekAlternateSyntaxesReturnTheSameResults" -Dtests.seed=CD667C34D6CA555A -Dtests.security.manager=false -Dtests.locale=mt-MT -Dtests.timezone=Iceland -Druntime.java=11

java.lang.AssertionError
	at __randomizedtesting.SeedInfo.seed([CD667C34D6CA555A:1584FB066FFA0A2B]:0)
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at org.opensearch.sql.sql.DateTimeFunctionIT.testWeekAlternateSyntaxesReturnTheSameResults(DateTimeFunctionIT.java:466)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
        ... (continued stack trace)
```

Reply:
Looking over this again I forgot to link the commit where this was added.
https://github.com/Bit-Quill/opensearch-project-sql/pull/175/commits/6a0c01326cfdecf491b45a0ee1e857d087822235

Reply:
```java
result1.similar(result2)
```
fails, because `resultX` contains entire JSON including column names which are different.
Try this instead:
```java
res1.getJSONArray("datarows").similar(res2.getJSONArray("datarows"))
```
And try tests with some data from `calcs`:
```sql
CAST(date0 AS date)            
datetime(CAST(time0 AS STRING))
CAST(time1 AS time)            
CAST(datetime0 AS timestamp)   
```
(cast as datetime is missing, so there is a workaround)

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/175/commits/c49c11270b09e3fb546d35fd8ebb9a816fa8881c